### ISSUE TITLE
optional support for initializer_list

### ISSUE THREAD

**Issue by 0xDEAD** [ndsev/zserio#36]
It would be great to add constructors that allow usage of C++11 initializer_list, at list for simple structures without options and for arrays.

Example:
```
struct Wgs84
{
  float32 longitude;
  float32 latitude;
}
```

generates a simple C++ class, that needs to be filled in your code like this:
```C++
Wgs84 coordinate;
coordinate.setLongitude(11.);
coordinate.setLatitude(50.);
```

It would be great if one could use:
```C++
Wgs84 coordinate(11., 50.);
```
or
```C++
Wgs84 coordinate({11., 50.});
```
instead.
The same is valid for arrays that contain simple types, like an array of integers or floats.

The most easy way to allow std::initizliser_list for sequences is probably to group the members into a structure and add an additional constructor that accepts a const reference to this (internal) struct. The rest will be done by the compiler automatically. Here is how the the generated class could look like (shortened):

```C++
class Wgs84
{
public:
    struct members
    {
        float32 longitude;
        float32 latitude;
    }
    Wgs84(const members& m) : m_members(m) {}

    [... generated methods as before ...]

private:
    members m_members;
}
```

**Comment by 0xDEAD** [comment]
Of course this feature could be optional (enabled with a parameter, disabled by default) to avoid "pollution with C++11-code" for legacy users of the generated code

**Comment by 0xDEAD** [comment]
Snippet for additional generated constructor:
```C++
ArrayExample::ArrayExample(const std::initializer_list<...>& list)
       : m_list(list.size())
        , m_numItems(list.size())
{
    m_list.assign(list.begin(), list.end());
}
```

**Comment by mikir** [comment]
We will try to keep this issue in mind during design of new C++11 emitter within issue #108.

**Comment by Mi-La** [comment]
Will be solved by field constructors implemented in #108.
